Allow development environment to work on arm based macOS machines

eaon opened the pull request:
# Description

Fixes #1591

This also changes the install instructions on macOS in general, unifying the setup for Intel and arm64 machines. This also drops instructions to use pyenv in favour of installing Python 3.9 via Homebrew. This is necessary to allow x86_64 mode Python 3.9 virtual environments to work as we need them to.

# Test Plan

- [x] Client runs after following instructions on a macOS machine with an Intel processor
- [x] Client runs after following instructions on a macOS machine with an arm64 processor

# Checklist

 - [x] These changes should not need testing in Qubes
 - [x] No update to the AppArmor profile is required for these changes
 - [x] No database schema changes are needed

eaon commented:
I don't think we currently have a developer with an M1/M2 setup - these instructions probably do not need visual verification there, so if there's some service with remote ssh access I think that would be sufficient to review this.

gonzalo-bulnes commented:
I think that @cfm has access to an Intel machine. @jdlrobson I think you do as well, don't you? If you feel like reviewing this PR, please do!

gonzalo-bulnes commented:
(I've been removing the `hackathon` labels in favor of `help wanted` and `good first issue` where relevant. I think they feel less event-specific and are more standard.)

zekehuntergreen commented on README.md:
```diff
@@ -135,14 +135,15 @@ socat TCP4-LISTEN:8081,fork,reuseaddr TCP4:A.B.C.D:8081
+    4. Follow the instructions at https://brew.sh/
+    5. Install the dependencies below in the same `x86_64` shell session
+3. Install dependencies via Homebrew: `brew install python@3.9 gnupg oath-toolkit`
 6. clone the SecureDrop Client repo and set up its virtual environment
```
Should the two following steps be steps 4 and 5?

eaon replied:
Yes, thanks!

zekehuntergreen commented on README.md:
```diff
@@ -135,14 +135,15 @@ socat TCP4-LISTEN:8081,fork,reuseaddr TCP4:A.B.C.D:8081
+    1. (Informational) If you already have a native `arm64` version of Homebrew, you will wind up with a separate Homebrew installation for each architecture. While they do not conflict, combining them in your `PATH` permanently may cause confusion about which brews are installed where. You may want to modify your `PATH` temporarily to perform this installation. The virtual environment you create ...
+    2. Run `/usr/sbin/softwareupdate --install-rosetta` to allow you to run `x86_64` binaries
+    3. Enter a shell in `x86_64` mode with `arch -x86_64 bash`
+    4. Follow the instructions at https://brew.sh/
```
I know all of these sub-items are part of "2. Install Homebrew" but this might be slightly clearer.
```suggestion
    4. Install Homebrew via the instructions at https://brew.sh/
```

zekehuntergreen reviewed (commented):
Thanks very much for this!
I'm on an M1 and can confirm that the client runs after following the instructions.

cfm reviewed (commented):
I can confirm that:

> - [x] Client runs after following instructions on a macOS machine with an Intel processor

`brew install [...]` was a no-op for me, but `rm -rf .venv && make venv-mac` succeeded.

I can approve and merge, @eaon, after you've reviewed @zekehuntergreen's suggestions.

cfm commented:
@eaon, if you have time to finalize this today or tomorrow, I can review and merge it!

eaon commented:
Added suggestions from @zekehuntergreen (thank you!), a small fix to work around the fact that some folks/environments might not have `which` installed, and rebased because `run.sh` got an update due to mac specific `mktemp` specifics being reused for tests. If CI works out, this would be ready to go @cfm :smile:

cfm reviewed (approved):
Looks good, and `rm -rf .venv && make venv-mac` still succeeds.  Thanks, @eaon!
